core/modules/flowgen.cc

```diff
@@ -282,8 +297,23 @@ CommandResponse FlowGen::CommandUpdate(const bess::pb::FlowGenArg &arg) {
     }
     template_size_ = arg.template_().length();
 
-    memset(templ_, 0, MAX_TEMPLATE_SIZE);
-    bess::utils::Copy(templ_, arg.template_().c_str(), template_size_);
+    const char *tmpl = arg.template_().c_str();
+    const Ethernet *eth = reinterpret_cast<const Ethernet *>(tmpl);
+    if (eth->ether_type != be16_t(Ethernet::Type::kIpv4)) {
+      return CommandFailure(EINVAL, "'template' is not IPv4");
+    }
+
+    const Ipv4 *ip = reinterpret_cast<const Ipv4 *>(eth + 1);
+    if (ip->protocol != Ipv4::Proto::kUdp and ip->protocol != Ipv4::Proto::kTcp) {
+      return CommandFailure(EINVAL, "'template' is not UDP or TCP");
+    }
+
+    if (l4_proto_ != ip->protocol) {
```

Comment:
Q: is there any reason to not change the L4 protocol of template?

Reply:
Just because current implementation doesn't differentiate new flows (with changed protocol) from the existing flows (old protocol)